File: backend/app/schemas/latex.py
from pydantic import BaseModel, Field


class LatexTableRequest(BaseModel):
    experiment_ids: list[int] = Field(default_factory=list)
    metric_names: list[str] = Field(default_factory=list)
    model_node_ids: list[int] = Field(default_factory=list)
    dataset_node_ids: list[int] = Field(default_factory=list)
    caption: str = "Main results."
    label: str = "tab:main-results"
    note: str | None = None
    placement: str = "t"
    precision: int = 2
    highlight_best: bool = True
    highlight_second: bool = True
    use_resizebox: bool = True
    compact: bool = True
    show_std: bool = True
    omit_zero_std: bool = True
    use_threeparttable: bool = True
    table_environment: str = "table"
    column_group_by: str = "dataset"


class LatexTableResponse(BaseModel):
    latex_code: str
    model_names: list[str]
    dataset_names: list[str]
    metric_names: list[str]
    num_runs: int
    num_metrics: int
    packages_hint: list[str]
    guideline_notes: list[str]

File: frontend/src/services/latex.ts
import { api } from './api'

export interface GenerateLatexPayload {
  experiment_ids: number[]
  metric_names?: string[]
  model_node_ids?: number[]
  dataset_node_ids?: number[]
  caption?: string
  label?: string
  note?: string
  placement?: string
  precision?: number
  highlight_best?: boolean
  highlight_second?: boolean
  use_resizebox?: boolean
  compact?: boolean
  show_std?: boolean
  omit_zero_std?: boolean
  use_threeparttable?: boolean
  table_environment?: string
  column_group_by?: string
}

export interface LatexTableResponse {
  latex_code: string
  model_names: string[]
  dataset_names: string[]
  metric_names: string[]
  num_runs: number
  num_metrics: number
  packages_hint: string[]
  guideline_notes: string[]
}

export async function generateLatexTable(payload: GenerateLatexPayload) {
  const { data } = await api.post<LatexTableResponse>('/latex/generate', payload)
  return data
}
